index.js

```diff
@@ -224,12 +224,21 @@ function Hook (modules, options, onrequire) {
       // Ex: require('foo/lib/../bar.js')
       // moduleName = 'foo'
       // fullModuleName = 'foo/bar'
-      if (hasWhitelist === true && modules.includes(moduleName) === false) {
-        if (modules.includes(fullModuleName) === false) return exports // abort if module name isn't on whitelist
+      let isWhitelistedSubmodule = false
+      if (hasWhitelist) {
+        // abort if module name isn't on whitelist
+        if (!modules.includes(moduleName) && !modules.includes(fullModuleName)) {
+          return exports
+        }
+
+        if (modules.includes(fullModuleName)) {
```

Comment:
```suggestion
        if (modules.includes(fullModuleName) && moduleName !== fullModuleName) {
```

This all looks good. I can't break it. :)

One thing that was niggling me a little bit is this one case. If `moduleName === fullModuleName` -- e.g. when doing `require('sub-module')` -- then this would result in (a) the `isWhitelistedSubmodule === true` being slightly misleading and (b) the `if (!isWhitelistedSubmodule) { ...` block below being skipped.  However, I can't come up with a case where skipping that block isn't just fine -- I expect `res === filename` every time that happens.

tl;dr: What you have here works, as best I can tell, but I think I'd prefer to have this small patch.

Reply:
Let me know if you have thoughts on the suggestion. Then I can get a release out of this soon.